### ISSUE TITLE
[Improvement] The Login API still gets called when providing an invalid formatted email concerns

### ISSUE THREAD

**Issue by minhnimble** [thienhm17/Nimble-Code-Chanllenge#11]
When providing an invalid email format string into the email text field, the Login button is still enabled and thus the login API still gets called and it always results in an error message. What is your opinion on checking for the validity of the email before calling the API as well?

<img src="https://user-images.githubusercontent.com/70877098/161497101-029e8e43-b47a-46bd-88f3-d0658b79e775.png" width="300">

At the same time, when trying to call the Login API without an internet connection, there is a weird message that mentions about some technical terms on the prefix part instead of just the actual case for the issue. As a result, from the user perspective, it is a bit confusing here for the user on the extra message. What is your opinion on this concern?

<img src="https://user-images.githubusercontent.com/70877098/161497968-070eea53-4a60-46c0-ac42-6c3771e68f29.PNG" width="300">

**Comment by thienhm17** [comment]
Yes, I think we should validate the email format and check if the email and password fields are filled or not before enabling the Login button. It will help the user avoid some mistake and some unexpected error messages.

Yes, we should let the user know if they are on connected with the internet or not. And the user should not be seen that development message as well.
